azurerm/resource_arm_storage_filesystem.go

```diff
@@ -0,0 +1,266 @@
+package azurerm
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	azauto "github.com/Azure/go-autorest/autorest/azure"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage"
+	"github.com/tombuildsstuff/giovanni/storage/2018-11-09/datalakestore/filesystems"
+)
+
+func resourceArmStorageFilesystem() *schema.Resource {
+	return &schema.Resource{
+		Create:        resourceArmStorageFilesystemCreate,
+		Read:          resourceArmStorageFilesystemRead,
+		Delete:        resourceArmStorageFilesystemDelete,
+		SchemaVersion: 1,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
```

Comment:
since we need to switch this over to using the Storage Account ID, rather than the Storage Account Name - we also need to add an import function to look up the Storage Account ID here.

As we don't have many examples of this I hope you don't mind but I'm going to push a commit to fix up the other changes and add this